### ISSUE TITLE
Nested calc with css variables

### ISSUE THREAD

**Issue by kangw3n** [postcss/postcss-calc#91]
*thanks @evilebottnawi  for redirection. [#816](https://github.com/cssnano/cssnano/issues/816#issue-492559307)*

I'm encountering an issues while using calc() with  CSS Custom variables where presets automatically remove my nested calc while custom variables is unable to resolve in build time.

Given that: 

```css
body {
  top: calc(100% - calc(10px + 10px));
}
/* will compile to*/
body {
  top: calc(100% - 20px);
}
```
when it comes to a css variables

```css
body {
  top: calc(var(--pic-height) - calc(var(--width-height) + var(--border-gap)));
}
/* will compile to*/
body {
  top: calc(var(--pic-height) - var(--width-height) + var(--border-gap));
}
```

![Capture](https://user-images.githubusercontent.com/6106265/64834625-34342280-d615-11e9-9d01-948b4a2ed36b.JPG)

which is incorrect in mathematical stand points. To solve this problem, I end up by setting **calc** preset to **false** in plugin options.

Is there any possible to omit any clearing unintended  nested calc while encountering css variable or if there is any better solution while dealing with variables value?

**Comment by mcfanng** [comment]
Yes I have the similar issue.
It seems as though when using nested vars in a calc with parentheses, the parentheses get removed which results in an incorrect calculation. Here is the case I am currently having in .scss file. 
Before Pack:

```css
body {
	--header-height: 57px;
	--header-bottom-padding: 8px;
	--footer-height: 38px;
	--footer-vertical-padding: 32px;
	--body-height: calc(100vh - (var(--footer-height) + var(--footer-vertical-padding)));
	--site-wrapper-min-height: calc(var(--body-height) - (var(--header-height) + var(--header-bottom-padding)));
	--calendar-body-height: calc(
		var(--site-wrapper-min-height) -
			(var(--month-header-height) + var(--days-header-height) + var(--site-content-children-vertical-padding))
	);
	--month-header-height: 30px;
	--days-header-height: 30px;
	--site-content-children-vertical-padding: 10px;
	width: 100%;
	height: var(--body-height);
	z-index: 10;
}
```
After pack: 

```css
body {
	--header-height: 57px;
	--header-bottom-padding: 8px;
	--footer-height: 38px;
	--footer-vertical-padding: 32px;
	--body-height: calc(100vh - var(--footer-height) - var(--footer-vertical-padding));
	--site-wrapper-min-height: calc(var(--body-height) - (var(--header-height) + var(--header-bottom-padding)));
	--calendar-body-height: calc(
		var(--site-wrapper-min-height) -
			var(--month-header-height) + var(--days-header-height) + var(--site-content-children-vertical-padding));
	--month-header-height: 30px;
	--days-header-height: 30px;
	--site-content-children-vertical-padding: 10px;
	width: 100%;
	height: var(--body-height);
	z-index: 10;
}
```
If you notice that `--body-height` value went from `calc(100vh - (var(--footer-height) + var(--footer-vertical-padding)))` to `calc(100vh - var(--footer-height) - var(--footer-vertical-padding))` which works just fine.
However `--calendar-body-height` value went from `calc(
		var(--site-wrapper-min-height) -
			(var(--month-header-height) + var(--days-header-height) + var(--site-content-children-vertical-padding)))` to `calc(
		var(--site-wrapper-min-height) -
			var(--month-header-height) + var(--days-header-height) + var(--site-content-children-vertical-padding))` which produces an incorrect result.

**Comment by abstractvector** [comment]
I raised an issue on cssnano (https://github.com/cssnano/cssnano/issues/628) but believe the issue lies upstream from there in this repo.

I am able to reproduce the same behavior, that the following two CSS styles produce different outputs (confirmed in Chrome 80 and Firefox 74 on Mac OS 10.15.3):

```css
padding-bottom: calc(100% / (var(--aspect-ratio)));
```
Computed: `padding-bottom: 276.906px` (correct)

<img width="656" alt="Screen Shot 2020-03-27 at 8 05 44 AM" src="https://user-images.githubusercontent.com/726531/77770049-e35ca800-7001-11ea-8c4a-9c4ad6b9280e.png">

```css
padding-bottom: calc(100% / var(--aspect-ratio));
```
Computed: `padding-bottom: 0px` (incorrect)

<img width="656" alt="Screen Shot 2020-03-27 at 8 05 28 AM" src="https://user-images.githubusercontent.com/726531/77770038-dfc92100-7001-11ea-941f-00ce9743c457.png">

I'm not sure why the parentheses matter in this situation, but the empirical evidence suggests they do. I suggest therefore the parentheses should _not_ be removed.

**Comment by dangelion** [comment]
Confirm the bug. Is there any way to solve?

**Comment by dangelion** [comment]
I solved removing postcss-calc and using this:
https://github.com/nico-jacobs/postcss-remove-nested-calc
Works like a charm. It'll be good if this library could be recognized as official postcss plugin. Or someone could be inspired from it to fix the bug in postcss-calc.

**Comment by ludofischer** [comment]
So the first issue is already fixed in 8.2, while the issue in https://github.com/postcss/postcss-calc/issues/91#issuecomment-605059796 is a duplicate of #113
